### ISSUE TITLE
move SIBaseUnits_t to trait UnitDimension

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/picongpu#1431]
We should move [SIBaseUnits](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/picongpu/include/simulation_defines/unitless/speciesAttributes.unitless#L41-L52) to the trait definition of [UnitDimension](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/picongpu/include/traits/UnitDimension.hpp) to allow more readable definitions.

with SIBaseUnits
```C++
std::vector<float_64> unitDimension( NUnitDimension, 0.0 );
unitDimension.at(traits::SIBaseUnits::length) =  1.0;
unitDimension.at(traits::SIBaseUnits::mass)   =  1.0;
unitDimension.at(traits::SIBaseUnits::time)   = -1.0;
```
without 
```C++
std::vector<float_64> unitDimension( NUnitDimension, 0.0 );
unitDimension.at(0) =  1.0;
unitDimension.at(1)   =  1.0;
unitDimension.at(2)   = -1.0;
```

**Comment by psychocoderHPC** [comment]
Maybe we should move `SIBaseUnits` out of the namespace `traits`

**Comment by ax3l** [comment]
oh I did not see this issue. Did I accidentally fix it with #1429 (see b62c6af9e56527aae5322efc7d8f75c04b4e182a) ? ;)

It's not technically a trait but fits their quite well for now.

**Comment by psychocoderHPC** [comment]
solved with #1429
